docs/hello_nextflow/03_hello_gatk.md

```diff
@@ -680,37 +565,99 @@ process GATK_JOINTGENOTYPING {
+
+Add this after the call to GATK_HAPLOTYPECALLER:
+
+```groovy title="hello-gatk.nf"
+// Collect variant calling outputs across samples
+all_vcfs = GATK_HAPLOTYPECALLER.out[0].collect()
+all_tbis = GATK_HAPLOTYPECALLER.out[1].collect()
+```
+
+!!! tip
+
+    You can view the contents of the channel after performing this collect operation using `.view()`
+
+#### 5.4. Add a call to the workflow block to run GATK_JOINTGENOTYPING
 
 ```groovy title="hello-gatk.nf"
 // Consolidate GVCFs and apply joint genotyping analysis
 GATK_JOINTGENOTYPING(
-    sample_map,
+    all_vcfs,
+    all_tbis,
     params.cohort_name,
-    params.genome_reference,
-    params.genome_reference_index,
-    params.genome_reference_dict,
-    params.calling_intervals
+    ref_file,
+    ref_index_file,
+    ref_dict_file,
+    intervals_file
 )
 ```
 
-#### 6.7. Add default value for the cohort name parameter up top
+#### 5.5. Run the workflow
+
+```bash
+nextflow run hello-gatk.nf -resume
+```
+
+Oh no! The pipeline produces an error. When we dig into the console output, we can see the command executed isn't correct:
+
+```bash
+Command executed:
+
+  gatk GenomicsDBImport -V reads_mother.bam.g.vcf reads_father.bam.g.vcf reads_son.bam.g.vcf --genomicsdb-workspace-path family_trio_gdb -L intervals.bed
+
+  gatk GenotypeGVCFs -R ref.fasta -V gendb://family_trio_gdb -O family_trio.joint.vcf -L intervals.bed
+```
+
+Can you spot the error? We gave `gatk GenomicsDBImport` multiple VCF files for a single `-V` argument, but the tool expects a separate `-V` argument for each VCF file.
+
+#### 5.6. Construct a command line with a separate `-V` argument for each input VCF
+
+We use some string manipulations to repeat the VCFs with the argument `-V`, and replace `-V ${vcfs}` with the resulting `${vcfs_line}`:
+
+_Before:_
 
 ```groovy title="hello-gatk.nf"
-// Base name for final output file
-params.cohort_name = "family_trio"
+    script:
+    """
+    gatk GenomicsDBImport \
+        -V ${vcfs} \
+        --genomicsdb-workspace-path ${cohort_name}_gdb \
+        -L ${interval_list}
 ```
 
-#### 6.8. Run the workflow to verify that it generates the final VCF output as expected
+_After:_
+
+```groovy title="hello-gatk.nf"
+    script:
+    def vcfs_line = vcfs.collect { "-V ${it}" }.join(' ')
+    """
+    gatk GenomicsDBImport \
+        ${vcfs_line} \
+        --genomicsdb-workspace-path ${cohort_name}_gdb \
+        -L ${interval_list}
+```
+
+#### 5.7. Run the workflow to verify that it generates the final VCF output as expected
 
 ```bash
-nextflow run hello-gatk.nf
+nextflow run hello-gatk.nf -resume
 ```
 
 Now we see the additional process show up in the log output (showing the compact view):
 
 ```console title="Output"
-N E X T F L O W  ~  version 23.10.1
+N E X T F L O W  ~  version 24.02.0-edge
```

Comment:
```suggestion
N E X T F L O W  ~  version 24.04.4
```

Reply:
noted, let's tackle this in a later pass.